asset/js/setup.js

```diff
@@ -1,7 +1,36 @@
 /* eslint-disable no-undef */
 
 Vue.use(VueStrap);
-const vm = new Vue({
-  el: '#app',
+
+const typeahead = VueStrap.components.typeahead;
+
+jQuery.getJSON('../../site.json').then((siteData) => {
+  const titleArray = jQuery.map(siteData.pages, object => object.title);
+  const routeArray = jQuery.map(siteData.pages, object => object.src);
+
+  const vm = new Vue({
+    el: '#app',
+    components: {
+      typeahead,
+    },
+    data() {
+      return {
+        searchData: titleArray,
+      };
+    },
+    methods: {
+      searchCallback(match) {
+        const index = titleArray.indexOf(match);
+        const route = routeArray[index];
+        window.location.pathname = route.replace('md', 'html');
```

Comment:
If the route contains `md` in the name, it will give an incorrect route.

Reply:
Using the path module, 
1. Check if `path.extname(route)` is `.md`
2. `path.join(path.dirname(route), path.basename(route, path.extname) + '.html')`

Reply:
Thanks! Will change it to replace the extensions instead of just 'md'.